### PR TITLE
update to 3.5.4 for security fixes.

### PR DESCRIPTION
**Pull request by ale8k**
## Description

Updates JIMM to 3.5.4 (exactly), this should fix some vulns present in our recent scans.

Juju is on 1.23 now, and as such the tidy updated our go version. 

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/canonical/jimm/v3
 
-go 1.22.5
+go 1.23
+
+toolchain go1.23.2
```
Comment (babakks) [inline go.mod]: The [`go.mod`](https://github.com/juju/terraform-provider-juju/blob/main/go.mod) file in the Juju Terraform Provider has two `require` clauses (both with direct dependencies), and puts the Juju module (with a comment pointing to the Juju version) in the first one. I think it's a bit nicer to do the same thing in JIMM:

```
module github.com/juju/terraform-provider-juju

go 1.21

require (
	github.com/bflad/tfproviderlint v0.30.0
	github.com/hashicorp/terraform-plugin-docs v0.19.4
	// v3.5.3
	github.com/juju/juju v0.0.0-20240724081236-63d460f9ee6c

)

require (
	github.com/canonical/jimm-go-sdk/v3 v3.0.4
	github.com/dustin/go-humanize v1.0.1
	github.com/hashicorp/terraform-json v0.22.1
    ...
```

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/canonical/jimm/v3
 
-go 1.22.5
+go 1.23
```
Comment (babakks) [inline go.mod:3]: Why not `go 1.23.2` and remove the `toolchain` statement?

File: go.mod (hunk 1 above)
Comment (ale8k) [reply go.mod]: I'm not understanding why?

File: go.mod (hunk 2 above)
Comment (ale8k) [reply go.mod:3]: I've removed it now aha, it was autoadded

File: go.mod (hunk 1 above)
Comment (babakks) [reply go.mod]: Just to have Juju near the top of the `go.mod` file, not in the middle the deps list. Anyway, this is just a suggestion, so feel free to ignore.

**Review by babakks — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (ale8k) [reply go.mod]: Done it!

**Review by SimoneDutto — APPROVED** [review]

**Pull request merged**